### ISSUE TITLE
Weird URLs are converted from '%' to '%25' in Markdown links

### ISSUE THREAD

**Issue by kaho2063** [microsoft/vscode#156859]
Issue Type: <b>Bug</b>

In the Markdown source code, when Ctrl + clicking on the URL part of the following description, "%" in the URL is converted to "%25". It doesn't happen when clicking on a link in the preview, or "Follow Link" that pops up.
``` markdown
[Rust Japan Wiki](https://rust-japan.game-info.wiki/d/%A5%C8%A5%C3%A5%D7%A5%DA%A1%BC%A5%B8)
[link-1]: https://rust-japan.game-info.wiki/d/%A5%C8%A5%C3%A5%D7%A5%DA%A1%BC%A5%B8
```

It does not occur in the URL only description or plain text mode as shown below.
``` text
https://rust-japan.game-info.wiki/d/%A5%C8%A5%C3%A5%D7%A5%DA%A1%BC%A5%B8
```

VS Code version: Code 1.69.2 (3b889b090b5ad5793f524b5d1d39fda662b96a2a, 2022-07-18T16:14:10.636Z)
OS version: Linux x64 5.4.0-122-generic
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i3-5005U CPU @ 2.00GHz (4 x 1817)|
|GPU Status|2d_canvas: unavailable_software<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>opengl: disabled_off<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software|
|Load (avg)|2, 1, 1|
|Memory (System)|7.69GB (0.35GB free)|
|Process Argv|--unity-launch --crash-reporter-id 05cb9321-6408-47b1-b4d1-ef314d862d3d --crash-reporter-id 05cb9321-6408-47b1-b4d1-ef314d862d3d --crash-reporter-id 05cb9321-6408-47b1-b4d1-ef314d862d3d|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|mate|
|XDG_CURRENT_DESKTOP|MATE|
|XDG_SESSION_DESKTOP|mate|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (8)</summary>

Extension|Author (truncated)|Version
---|---|---
githistory|don|0.6.19
EditorConfig|Edi|0.16.4
vscode-macros|EXC|1.4.0
vscode-language-pack-ja|MS-|1.69.7130911
python|ms-|2022.10.1
vscode-pylance|ms-|2022.7.40
googlesearch|nat|0.1.0
markdown-all-in-one|yzh|3.4.3

</details>

**Comment by kaho2063** [comment]
Thank you for your response!

2022年8月10日(水) 0:42 Matt Bierner ***@***.***>:

**Comment by meganrogge** [comment]
The issue still reproduces for me

**Comment by rebornix** [comment]
I'm still seeing this in today's Insiders.

**Comment by mjbvz** [comment]
Hm yes only seems fixed on the second link (the definition). Will fix  see about full fix in October

**Comment by mjbvz** [comment]
Actually this is blocked by #85930 afaik. The server is returning the correct uri string, but `vscode.Uri.parse` mangles the uri. I also tried switching to `vscode.Uri.from`, but still could not construct the correct uri. Something always ends up being double encoded 

Let me know if you have any workarounds @jrieken

**Comment by jrieken** [comment]
Yeah, there is unfortunately no workaround and allowing `URL` would be the best way going forward

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!
